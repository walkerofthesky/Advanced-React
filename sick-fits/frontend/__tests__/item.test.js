import { shallow } from 'enzyme';

import Item from '../components/item';

const fakeItem = {
  id: 'fakeid',
  title: 'Something Cool',
  price: 5000,
  description: 'Something you wish you had, like a pet turtle.',
  image: 'felix.gif',
  largeImage: 'garfield.png'
};

describe('<Item />', () => {
  it('renders the image properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders and displays properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);

    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$50');

    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders the buttons properly', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
