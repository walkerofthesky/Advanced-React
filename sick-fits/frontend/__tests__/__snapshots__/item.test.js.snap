// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Something Cool"
    src="felix.gif"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "fakeid",
          },
        }
      }
    >
      <a>
        Something Cool
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    Something you wish you had, like a pet turtle.
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "fakeid",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="fakeid"
    />
    <DeleteItem
      id="fakeid"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
